File: cypress/cypress/support/pages/CheckoutPage.ts
export class CheckoutPage {
  fillShipping() {
    cy.get('[data-testid="checkout-address"]').clear().type("412 Hood Street");
    cy.get('[data-testid="checkout-city"]').clear().type("Portland");
    cy.get('[data-testid="checkout-region"]').clear().type("OR");
    cy.get('[data-testid="checkout-postal"]').clear().type("97214");
  }

  placeOrder() {
    cy.get('[data-testid="place-order"]').click();
  }
}

File: selenium-csharp/Kestrel.Automation/Pages/CheckoutPage.cs
namespace Kestrel.Automation.Pages;

public class CheckoutPage
{
    private readonly Func<string, IWebElement> _el;

    public CheckoutPage(Func<string, IWebElement> el)
    {
        _el = el;
    }

    public void FillShipping(string address = "412 Hood Street", string city = "Portland", string region = "OR", string postal = "97214")
    {
        Type("checkout-address", address);
        Type("checkout-city", city);
        Type("checkout-region", region);
        Type("checkout-postal", postal);
    }

    public void PlaceOrder() => _el("place-order").Click();

    private void Type(string testId, string value)
    {
        var field = _el(testId);
        field.Clear();
        field.SendKeys(value);
    }
}
